reflex/app.py

```diff
@@ -1067,50 +1067,60 @@ async def process(
         headers: The client headers.
         client_ip: The client_ip.
 
+    Raises:
+        ReflexError: If a reflex specific error occurs during processing the event.
+
     Yields:
         The state updates after processing the event.
     """
-    # Add request data to the state.
-    router_data = event.router_data
-    router_data.update(
-        {
-            constants.RouteVar.QUERY: format.format_query_params(event.router_data),
-            constants.RouteVar.CLIENT_TOKEN: event.token,
-            constants.RouteVar.SESSION_ID: sid,
-            constants.RouteVar.HEADERS: headers,
-            constants.RouteVar.CLIENT_IP: client_ip,
-        }
-    )
-    # Get the state for the session exclusively.
-    async with app.state_manager.modify_state(event.substate_token) as state:
-        # re-assign only when the value is different
-        if state.router_data != router_data:
-            # assignment will recurse into substates and force recalculation of
-            # dependent ComputedVar (dynamic route variables)
-            state.router_data = router_data
-            state.router = RouterData(router_data)
-
-        # Preprocess the event.
-        update = await app._preprocess(state, event)
-
-        # If there was an update, yield it.
-        if update is not None:
-            yield update
-
-        # Only process the event if there is no update.
-        else:
-            if app._process_background(state, event) is not None:
-                # `final=True` allows the frontend send more events immediately.
-                yield StateUpdate(final=True)
-                return
+    from reflex.utils import telemetry
+    from reflex.utils.exceptions import ReflexError
```

Comment:
i've seen this pattern a lot... is there a circular dep when trying to import reflex.utils at the top of the module?

Reply:
Yeah, at least one of the files, import at the top causes circular dependency